File: CPP/Program/Motor.cpp
/*Saya Legi Kuswandi mengerjakan LP4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang 
telah dispesifikasikan. Aamiin.*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

class Motor : public Vehicle
{
private:
    string jenis_motor;
    string kapasitas_tanki;
public:
    // Constructor
    Motor()
    {

    }

    // Constructor : Overloading
    Motor(string jenis_motor, string platnomor, string merk, string tahun_produksi, string warna) : Vehicle(platnomor, merk, tahun_produksi, warna){
        this->jenis_motor = jenis_motor;
        kapasitas_tanki = "";
    }

    Motor(string jenis_motor, string kapasitas_tanki, string platnomor, string merk, string tahun_produksi, string warna) : Vehicle(platnomor, merk, tahun_produksi, warna){
        this->jenis_motor = jenis_motor;
        this->kapasitas_tanki = kapasitas_tanki;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Motor
    void setJenis_motor(string jenis_motor){
        this->jenis_motor = jenis_motor;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Motor
    string getJenis_motor(){
        return jenis_motor;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Motor
    void setKapasitas_tanki(string kapasitas_tanki){
        this->kapasitas_tanki = kapasitas_tanki;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Motor
    string getKapasitas_tanki(){
        return kapasitas_tanki;
    }
    
    // Destructor
    ~Motor()
    {

    }
};

File: Python/Program/Motor.py
from Vehicle import Vehicle

class Motor(Vehicle):
    def __init__(self, jenis_motor="", kapasitas_tangki="", platnomor="", merk="", tahun_produksi="", warna=""):
        # Memanggil konstruktor kelas induk (Vehicle)
        super().__init__(platnomor, merk, tahun_produksi, warna)
        # Inisialisasi atribut khusus motor
        self.jenis_motor = jenis_motor  # Jenis motor (Matic, Manual, dsb.)
        self.kapasitas_tangki = kapasitas_tangki  # Kapasitas tangki bahan bakar

    # Setter untuk jenis motor
    def set_jenis_motor(self, jenis_motor):
        self.jenis_motor = jenis_motor

    # Getter untuk jenis motor
    def get_jenis_motor(self):
        return self.jenis_motor

    # Setter untuk kapasitas tangki bahan bakar
    def set_kapasitas_tangki(self, kapasitas_tangki):
        self.kapasitas_tangki = kapasitas_tangki

    # Getter untuk kapasitas tangki bahan bakar
    def get_kapasitas_tangki(self):
        return self.kapasitas_tangki
